pkg/controllers/deprovisioning/consolidation.go

```diff
@@ -58,58 +71,97 @@ func (c *Consolidation) ShouldDeprovision(_ context.Context, n *state.Node, prov
 	return provisioner != nil && provisioner.Spec.Consolidation != nil && ptr.BoolValue(provisioner.Spec.Consolidation.Enabled)
 }
 
-// sortCandidates orders deprovisionable nodes by the disruptionCost
-func (c *Consolidation) SortCandidates(nodes []CandidateNode) []CandidateNode {
-	sort.Slice(nodes, func(i int, j int) bool {
-		return nodes[i].disruptionCost < nodes[j].disruptionCost
-	})
-	return nodes
-}
-
 // computeCommand generates a deprovisioning command given deprovisionable nodes
-func (c *Consolidation) ComputeCommand(ctx context.Context, attempt int, candidates ...CandidateNode) (Command, error) {
+func (c *Consolidation) ComputeCommand(ctx context.Context, candidates ...CandidateNode) (Command, error) {
 	// the last cluster consolidation wasn't able to improve things and nothing has changed regarding
 	// the cluster that makes us think we would be successful now
 	if c.lastConsolidationState == c.cluster.ClusterConsolidationState() {
 		return Command{action: actionDoNothing}, nil
 	}
+	// First sort the candidate nodes by disruption cost.
+	sort.Slice(candidates, func(i int, j int) bool {
+		return candidates[i].disruptionCost < candidates[j].disruptionCost
+	})
+
+	// We use this timestamp for the Consolidation TTL so that our TTL logic waits one time per consolidation loop.
+	validationTimestamp := &time.Time{}
+
 	// First delete any empty nodes we see
-	if attempt == 0 {
-		if cmd := c.deleteEmpty(ctx, candidates...); cmd.action == actionDelete {
-			return cmd, nil
-		}
+	cmd, err := c.deleteEmpty(ctx, validationTimestamp, candidates...)
+	if err != nil {
+		return Command{action: actionFailed}, fmt.Errorf("deleting empty nodes for consolidation, %w", err)
```

Comment:
Just a general question: why do we need to return an actionFailed? Why not just an empty action? Shouldn't the error communicate the failure?

Reply:
I disagree, reconcile.Result{} returns and empty result and an error together, which represents failure of an action.

Reply:
Discussed offline, continuing to punt this refactor to a future PR.